dialogs.c

```diff
@@ -914,18 +914,20 @@ gint create_blocking_popup(GtkMessageType type, GtkButtonsType button,
 static glong date_compare_against_build_date(const char *iso8601_date)
 {
 #ifdef GIT_COMMIT_TIMESTAMP
-	GTimeVal time;
 	glong build_time = atol(GIT_COMMIT_TIMESTAMP);
-	gboolean parsed;
 	glong ret = 0;
 
-	parsed = g_time_val_from_iso8601(iso8601_date, &time);
-	if (parsed) {
-		ret = time.tv_sec - build_time;
-	} else {
+	GDateTime *time = g_date_time_new_from_iso8601(iso8601_date, NULL);
+	if (time == NULL) {
 		printf("%s could not parse date. Not a ISO 8601 format.", __func__);
+		return -1;
```

Comment:
returning -1 here will lead to a dialog saying `This software is newer than the latest release`
maybe the codepaths could be fixed where this is called; or just return 0 ?